test/gtest/uct/test_mem.cc

```diff
@@ -106,25 +108,60 @@ UCS_TEST_P(test_mem, md_alloc) {
         status = uct_md_query(md, &md_attr);
         ASSERT_UCS_OK(status);
 
-        for (nonblock = 0; nonblock <= 1; ++nonblock) {
-            int flags = nonblock ? UCT_MD_MEM_FLAG_NONBLOCK : 0;
+        for (mem_type = UCS_MEMORY_TYPE_HOST; mem_type < UCS_MEMORY_TYPE_LAST; mem_type++) {
+            params.mem_type = (ucs_memory_type_t)mem_type;
+            if (md_attr.cap.alloc_mem_types & UCS_BIT(mem_type)) {
+                if (mem_type == UCS_MEMORY_TYPE_HOST) {
+                    for (nonblock = 0; nonblock <= 1; ++nonblock) {
+                        int flags = nonblock ? UCT_MD_MEM_FLAG_NONBLOCK : 0;
 
-            flags         |= UCT_MD_MEM_ACCESS_ALL;
-            params.flags   = flags;
-            params.mds.mds = &md;
+                        flags         |= UCT_MD_MEM_ACCESS_ALL;
+                        params.flags   = flags;
+                        params.mds.mds = &md;
 
-            status = uct_mem_alloc(length, methods, 3, &params, &mem);
-            ASSERT_UCS_OK(status);
+                        status = uct_mem_alloc(length, methods, 3, &params, &mem);
+                        ASSERT_UCS_OK(status);
 
-            if (md_attr.cap.flags & UCT_MD_FLAG_ALLOC) {
-                EXPECT_EQ(UCT_ALLOC_METHOD_MD, mem.method);
-            } else {
-                EXPECT_NE(UCT_ALLOC_METHOD_MD, mem.method);
-            }
+                        if (md_attr.cap.flags & UCT_MD_FLAG_ALLOC) {
+                            EXPECT_EQ(UCT_ALLOC_METHOD_MD, mem.method);
+                        } else {
+                            EXPECT_NE(UCT_ALLOC_METHOD_MD, mem.method);
+                        }
+
+                        check_mem(mem, min_length);
+
+                        uct_mem_free(&mem);
+                    }
+                } else {
+                    if ((mem_type == UCS_MEMORY_TYPE_CUDA_MANAGED) ||
+                        (mem_type == UCS_MEMORY_TYPE_CUDA)) {
+                        int flags = UCT_MD_MEM_ACCESS_ALL;
+
+                        if (!mem_type_ctx_created) {
+                            address = mem_buffer::allocate(1, (ucs_memory_type_t)mem_type);
+                            mem_buffer::release(address, (ucs_memory_type_t)mem_type);
+                            mem_type_ctx_created = 1;
```

Comment:
why do you need this if you initialized in the init()

Reply:
Removed